util/aws_tools.go

```diff
@@ -156,3 +165,142 @@ func ConstructCloudwatchInsightsQueryDetail(entity cloudwatchlogsTypes.QueryDefi
 	queryDetailBuilder.WriteString(awsOuterEscape("))"))
 	return strings.ReplaceAll(queryDetailBuilder.String(), "%", "$")
 }
+
+func GenerateLogoutLoginUrl(destinationUrl string) url.URL {
+	loginUrlParams := url.Values{}
+	loginUrlParams.Add("Action", "login")
+	loginUrlParams.Add("Issuer", "aws-vault")
+	loginUrlParams.Add("SigninToken", SigninToken)
+	loginUrlParams.Add("Destination", destinationUrl)
+
+	loginUrl := url.URL{
+		Scheme:   "https",
+		Host:     "us-east-1.signin.aws.amazon.com",
+		Path:     "/federation",
+		RawQuery: url.Values(loginUrlParams).Encode(),
+	}
+
+	logoutUrlParams := url.Values{}
+	logoutUrlParams.Add("Action", "logout")
+	logoutUrlParams.Add("redirect_uri", loginUrl.String())
+
+	finalUrl := url.URL{
+		Scheme:   "https",
+		Host:     "us-east-1.signin.aws.amazon.com",
+		Path:     "/oauth",
+		RawQuery: url.Values(logoutUrlParams).Encode(),
+	}
+
+	if err := validateAWSUrl(finalUrl.String()); err != nil {
+		panic(err)
+	}
```

Comment:
because most of these are hardcoded values that are specific to the function body, it feels like we should be validating the input (that is, `destinationUrl`) earlier on. as-is, this is the surface area for the validation is larger than it needs to be